File: Problem-059/solution.js
function XORDecryption(cipher) {

    let bestSum = 0;
    let bestScore = -1;

    for (let a = 97; a <= 122; a++) {
        for (let b = 97; b <= 122; b++) {
            for (let c = 97; c <= 122; c++) {

                const key = [a, b, c];

                let text = "";

                for (let i = 0; i < cipher.length; i++) {

                    const decrypted =
                        cipher[i] ^ key[i % 3];

                    text += String.fromCharCode(decrypted);
                }
                let score =
                    (text.match(/ the /gi) || []).length +
                    (text.match(/ and /gi) || []).length +
                    (text.match(/ of /gi) || []).length;

                if (score > bestScore) {
                    bestScore = score;

                    bestSum = [...text]
                        .reduce(
                            (sum, ch) => sum + ch.charCodeAt(0),
                            0
                        );
                }
            }
        }
    }

    return bestSum;
}

File: Problem-059/solution.py
def xor_decryption(cipher):

    best_sum = 0
    best_score = -1

    for a in range(97, 123):
        for b in range(97, 123):
            for c in range(97, 123):

                key = [a, b, c]

                text = ""

                for i in range(len(cipher)):

                    decrypted = cipher[i] ^ key[i % 3]

                    text += chr(decrypted)

                score = (
                    text.lower().count(" the ") +
                    text.lower().count(" and ") +
                    text.lower().count(" of ")
                )

                if score > best_score:

                    best_score = score

                    best_sum = sum(ord(ch) for ch in text)

    return best_sum
